src/palace/manager/celery/tasks/overdrive.py

```diff
@@ -0,0 +1,328 @@
+    bind=True,
+    max_retries=4,
+    autoretry_for=(BadResponseException, RequestTimedOut),
+    throws=(RemoteIntegrationException,),
+    retry_backoff=60,
+)
+def import_collection_group(
+    task: Task,
+    collection_id: int,
+    *,
+    import_all: bool = False,
+    modified_since: datetime.datetime | None = None,
+    start_time: datetime.datetime | None = None,
+) -> None:
+    """Import an Overdrive collection and all its child (Advantage) collections.
+
+    This task orchestrates the import of a parent Overdrive collection and chains
+    the import of any associated Overdrive Advantage (child) collections. It uses
+    Celery's chord pattern to run child imports in parallel after the parent completes,
+    and then cleans up the shared identifier set.
+
+    Workflow:
+        1. Imports the parent collection
+        2. Passes the parent's identifier set to child collections
+        3. Imports all child collections in parallel (if any exist)
+        4. Cleans up the shared identifier set after all imports complete
+
+    :param collection_id: The ID of the parent collection to import
+    :param import_all: If True, import all titles regardless of change status.
+                       If False, only import changed titles based on modified_since.
+    :param modified_since: Only process titles modified after this datetime.
+                          See import_collection docstring for detailed behavior.
+    :param start_time: The datetime when this import began. Used to update the
+                      collection's timestamp. If None, uses current time.
+
+    .. note::
+       This task does not return a value. Results are tracked through the
+       linked chord and cleanup tasks.
+    """
+
+    import_collection.s(
+        collection_id=collection_id,
+        import_all=import_all,
+        page=None,
+        parent_identifiers=None,
+        return_identifiers=True,
+        modified_since=modified_since,
+        start_time=start_time,
+    ).apply_async(
+        link=import_children_and_cleanup_chord.s(
+            collection_id=collection_id,
+            import_all=import_all,
+            modified_since=modified_since,
+        )
+    )
+
+
+def rehydrate_identifier_set(
+    task: Task, identifier_set_info: dict[str, Any]
+) -> IdentifierSet:
+    """Reconstruct an IdentifierSet from its serialized representation.
+
+    This helper function takes a dictionary containing identifier set metadata
+    (specifically the Redis key) and recreates the IdentifierSet object that
+    can be used to access the data in Redis.
+
+    :param task: The Celery task instance (provides access to Redis client)
+    :param identifier_set_info: Dictionary containing the identifier set's key
+                                Format: {"key": ["key", "parts"]}
+    :return: Reconstructed IdentifierSet connected to Redis
+    """
+    return IdentifierSet(task.services.redis().client(), identifier_set_info["key"])
```

Comment:
This drops the identifiers sets expiration time. Everywhere else we do this operation this function would look like this:


```suggestion
def rehydrate_identifier_set(
    task: Task, identifier_set_info: RedisSetKwargs
) -> IdentifierSet:
    """Reconstruct an IdentifierSet from its serialized representation.

    This helper function takes a dictionary containing identifier set metadata
    (specifically the Redis key) and recreates the IdentifierSet object that
    can be used to access the data in Redis.

    :param task: The Celery task instance (provides access to Redis client)
    :param identifier_set_info: Dictionary containing the identifier set's key
                                Format: {"key": ["key", "parts"]}
    :return: Reconstructed IdentifierSet connected to Redis
    """
    return IdentifierSet(task.services.redis().client(), **identifier_set_info)
```

If you are going to make a helper function for this, maybe move it out to the appropriate identifier set file, and refactor the other code that is creating identifier sets to use it. It gets messy if every task decides to make its own helper functions for things like this.